### PR TITLE
issue94 open source event stream

### PR DESCRIPTION
**Pull request by leeduan**
Convert DCC middleware to using the open source event stream

### CODE REVIEW

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/TxRequestView.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ open class TxRequestViewEntityClass : UUIDEntityClass<TxRequestViewRecord>(TRV)
     fun findByTxHash(txHash: String) = find { TRV.txHash eq txHash }
 
     fun findExpired() = find {
-        (TRV.created lessEq OffsetDateTime.now().minusSeconds(30))
+        (TRV.created lessEq OffsetDateTime.now().minusSeconds(120))
```
Comment (leeduan) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/TxRequestView.kt:31]: should probably have this be a multiple of `blocksBeforeTimeout`

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/config/AppConfig.kt [hunk 2]
```diff
@@ -61,27 +52,8 @@ class AppConfig : WebMvcConfigurer {
...
-                    .readTimeout(60, TimeUnit.SECONDS) // ~ 12 pbc block cuts
-                    .build()
-                    .newWebSocketFactory("${node.scheme}://${node.host}:${node.port}/websocket")
-            )
-            .addMessageAdapterFactory(MoshiMessageAdapter.Factory())
-            .addStreamAdapterFactory(RxJava2StreamAdapterFactory())
-    }
-
-    @Bean
-    fun rpcClient(mapper: ObjectMapper, eventStreamProperties: EventStreamProperties): RpcClient =
```
Comment (mwoods-figure) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/config/AppConfig.kt:83]: Nice to see this gone

**Review by piercetrey-figure — APPROVED** [review]

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamConsumer.kt [hunk 3]
```diff
@@ -96,25 +108,33 @@ class EventStreamConsumer(
...
-                    burns = batch.burns(provenanceProperties.contractAddress),
-                    markerTransfers = batch.markerTransfers(),
-                )
 
-                transaction { EventStreamRecord.update(coinMovementEventStreamId, batch.height) }
-            }
+        runBlocking {
+            val netAdapter = okHttpNetAdapter(eventStreamUri)
+
+            log.info("Starting coin movement event stream at height $lastHeight")
```
[the remark this answers is not in the document]
Comment (leeduan) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamConsumer.kt:113]: we had to reprocess a bunch of reporting records at some point so had it split up then; not sure its needed anymore

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/CoinMovementMonitor.kt [hunk 4]
```diff
@@ -23,7 +23,7 @@ class CoinMovementMonitor(
 ) {
     private val log = logger()
 
-    private val eventStreamId = UUID.fromString(eventStreamProperties.id)
+    private val eventStreamId = UUID.fromString(eventStreamProperties.coinMovementId)
```
Comment (leeduan) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/CoinMovementMonitor.kt:26]: this was set wrongly so fixes existing bug

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/ExpiredEventReaper.kt [hunk 5]
```diff
@@ -31,7 +40,8 @@ class ExpiredEventReaper(private val pbcService: PbcService, private val txReque
                 response.txResponse.isSuccess() -> txRequestService.completeTxns(txHash)
                 response.txResponse.isFailed() -> {
                     log.error("Unexpected error for tx:$txHash, log:${response.txResponse.rawLog}")
-                    txRequestService.resetTxns(txHash)
+                    // TODO - determine if event stream error handling will catch all errors so this is no longer needed
```
Comment (ranadewall) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/ExpiredEventReaper.kt:43]: saw a pr go up for this today

**Review by ranadewall — APPROVED** [review]

**Pull request merged**